### DISCUSSION TITLE
HLS-Streaming stopped working

### DISCUSSION THREAD

**Discussion by nji9nji9** [aclap-dev/video-downloadhelper#44] [Help and Q&A]
VD worked fine with me - until I installed the newest version 8.1.4.0
and the CoApp vdhcoapp-2.0.10-win7-x86_64-installer (@ Win7-64).

Since then HLS-Streaming doesn't work anymore but always stops with an error,
at every site, both audio and video media.
Non-HLS-Streaming works.

The error text (using the YT page you mentioned at https://www.downloadhelper.net/howto:
https://www.youtube.com/watch?v=UDm8zTlP-mw

Error message:

--------------------------

GrabInfo: Cannot get info from https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1707271404/ei/jJDCZe7DIe-Dp-oP1v2fwAM/ip/37.4.229.32/id/5039bccd394ffa6c/itag/232/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D9347735%3Bdur%3D120.233%3Bgir%3Dyes%3Bitag%3D136%3Blmt%3D1689233178282447/rqh/1/hls_chunk_host/rr5---sn-4g5edndk.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/H3/mm/31,26/mn/sn-4g5edndk,sn-f5f7lnl7/ms/au,onr/mv/m/mvi/5/pl/23/initcwndbps/1791250/vprv/1/playlist_type/DVR/dover/13/txp/1432434/mt/1707249200/fvip/1/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,vprv,playlist_type/sig/AJfQdSswRQIhAIL23NbAAwz3aWMDufQ7Brhfkrrp-Tc5h0-8WhYh56PUAiB_9LDFgjE6xhlbf9rwD75i1wBFNrl7qPlqxc_jUdzsvQ%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRAIgDlNhBgatuDeIqrWs91lyHc0c7G5X_RxESoMB4ex7_iACIEtiBxxWfPmTuVVYBPq8-txmUZ8uBHLmPzuSlCowImnw/playlist/index.m3u8:Exit code: 1 ffprobe version n6.0 Copyright (c) 2007-2023 the FFmpeg developers built with gcc 10-win32 (GCC) 20220113 configuration: --arch=x86_64 --target-os=mingw32 --cross-prefix=x86_64-w64-mingw32- --extra-libs= --extra-ldflags='-static -static-libgcc -static-libstdc++' --extra-ldexeflags='-static -static-libgcc -static-libstdc++' --pkg-config-flags=--static --disable-autodetect --prefix=/ffmpeg-static/dist/windows/x86_64//ffmpeg --enable-version3 --pkg-config=pkg-config --enable-runtime-cpudetect --enable-pthreads --disable-w32threads --enable-libtheora --enable-libvpx --enable-libvorbis --disable-indev=sndio --disable-outdev=sndio --enable-neon --enable-libopenjpeg --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-openssl --enable-libvo-amrwbenc --enable-libopus --enable-libx264 --enable-libx265 --enable-libaom --enable-libwebp --enable-libxvid --enable-libmp3lame --enable-zlib --enable-ffprobe --enable-pic --disable-doc libavutil 58. 2.100 / 58. 2.100 libavcodec 60. 3.100 / 60. 3.100 libavformat 60. 3.100 / 60. 3.100 libavdevice 60. 1.100 / 60. 1.100 libavfilter 9. 3.100 / 9. 3.100 libswscale 7. 1.100 / 7. 1.100 libswresample 4. 10.100 / 4. 10.100 libpostproc 57. 1.100 / 57. 1.100 [tcp @ 000000000041ff00] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1707271404/ei/jJDCZe7DIe-Dp-oP1v2fwAM/ip/37.4.229.32/id/5039bccd394ffa6c/itag/232/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D9347735%3Bdur%3D120.233%3Bgir%3Dyes%3Bitag%3D136%3Blmt%3D1689233178282447/rqh/1/hls_chunk_host/rr5---sn-4g5edndk.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/H3/mm/31,26/mn/sn-4g5edndk,sn-f5f7lnl7/ms/au,onr/mv/m/mvi/5/pl/23/initcwndbps/1791250/vprv/1/playlist_type/DVR/dover/13/txp/1432434/mt/1707249200/fvip/1/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,vprv,playlist_type/sig/AJfQdSswRQIhAIL23NbAAwz3aWMDufQ7Brhfkrrp-Tc5h0-8WhYh56PUAiB_9LDFgjE6xhlbf9rwD75i1wBFNrl7qPlqxc_jUdzsvQ%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRAIgDlNhBgatuDeIqrWs91lyHc0c7G5X_RxESoMB4ex7_iACIEtiBxxWfPmTuVVYBPq8-txmUZ8uBHLmPzuSlCowImnw/playlist/index.m3u8: Error number -138 occurred

moz-extension://37888cde-17b9-480d-8fd6-0b63403c4731/background/main.js:31
cn</grabInfo/<@moz-extension://37888cde-17b9-480d-8fd6-0b63403c4731/background/main.js:31:88385

--------------------------

Any help appreciated.

### Addon & CoApp version and details

version: 8.1.4.0
beta: false
build date: 2024-01-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: de
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelperCoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "win7",
    "arch": "x86_64",
    "node": "10"
  },
  "home": "C:\\Users\\Guido",
  "converterBinary": "C:\\Program Files\\DownloadHelperCoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by paulrouget** [comment]
Do you mind giving the v8.2 a try (it's a beta): https://www.downloadhelper.net/beta

**Comment by nji9nji9** [comment]
Hi Paul.
Just tried... but: it's getting worse.
Now (at the above YT source) both HLS and non-HLS gives the error:

SideDownload error:

[tcp @ 00000000005072c0] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred

https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1707318258/ei/kkfDZbuoA7vD6dsPlImigA8/ip/37.4.228.227/id/5039bccd394ffa6c/itag/231/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D3521368%3Bdur%3D120.233%3Bgir%3Dyes%3Bitag%3D135%3Blmt%3D1689233179779963/rqh/1/hls_chunk_host/rr5---sn-4g5edndk.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/H3/mm/31,29/mn/sn-4g5edndk,sn-4g5lzner/ms/au,rdu/mv/m/mvi/5/pl/23/initcwndbps/2078750/vprv/1/playlist_type/DVR/dover/13/txp/1432434/mt/1707295970/fvip/5/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,vprv,playlist_type/sig/AJfQdSswRgIhALqT_3Kz9r-Bp8lei9eogUgraa2RWNv5eCt5p3f6FP4fAiEA9PpLr_4HhfnSAVxrHa7uhGp810kWisFd8xgHWtvDbe0%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRgIhAN5YzIL0lpLcpPh4H-oA81zhUXi4lGljKb7nVRHLetBxAiEAtwit6v2kPFZcFRTmv_ottNEWnvonGSX5mPRzvGXpRok%3D/playlist/index.m3u8: Error number -138 occurred

**Comment by paulrouget** [comment]
That seems like an antivirus blocking the coapp.

You use Windows 7?
What's your antivirus?

**Comment by nji9nji9** [comment]
It's MS Security Essentials.
I excluded CoApp EXE from its process list... no difference.
I do also have a [firewall ](https://github.com/henrypp/simplewall)- but coapp has full internet access there. 
....
If anything blocks the coapp... why hasn't it blocked non-hls with non-beta (see first post)?

EDIT
I just checked again right now.
Reinstall 8.1 official release.
Downloading hls version doesn't do (**no pop-up box**, just the error).
Downloading the non-hls version pops up the box ("Browser or CoApp").
Downloads fine.

EDIT2
Reinstalled an old version of VD (7.6) and VdhCoApp (1.6.3).
Tried on a different (non-YT) source: HLS worked there.

**Reply by GrampaWildWilly** [reply]
There's 3 .exe files in the CoApp directory that need to be made exceptions in the antivirus as well as the firewall.  It seems you are getting the same problem as a number of other folks.  Here for example: https://github.com/aclap-dev/video-downloadhelper/discussions/46.  It's simplest to just tell the antivirus & the firewall that everything in the CoApp directory is safe.  This is your antivirus/firewall throwing up a false positive.  Just do what is always necessary when new software gets this kind of false positive: Tell your antivirus & your firewall that it is triggering a false positive.  Make exceptions for the whole directory.  The vendor will eventually fix the false positive.  Well, I don't know.  Windows 8 isn't Microsoft's pet any more.  They may never fix it.  I can tell you that the default antivirus built into Windows 11, which I am running, has not kicked up any false positives on the CoApp.  I know you don't want hear that your solution is to upgrade your operating system.  That would infuriate me, so I'm not suggesting it to you.  I'm just saying Microsoft has managed to provide an antivirus/firewall that doesn't complain about the CoApp.  It's just not the software available on your system.  I suppose you could try to contact Microsoft Windows 8 support & tell them about their false positive.  In your shoes, I wouldn't bother.  I would just tell the software that it's generating a false positive.  It's simplest to just tell it about the whole directory than tell it about the pieces inside there.

**Reply by paulrouget** [reply]
> If anything blocks the coapp... why hasn't it blocked non-hls with non-beta?

Because your older coapp was whitelisted. It takes time for antivirus to catchup.

> I excluded CoApp EXE from its process list

Also allow ffprobe.exe and ffmpeg.exe from the same directory.

**Reply by GrampaWildWilly** [reply]
Plus the older CoApp didn't use ffmpeg for the downloads.  For other things, but not the downloads.

**Comment by nji9nji9** [comment]
It's as always: ... Grampa knows best :)

The culprit at me wasn't the AntiVirus (MS Security Essentials), but the [firewall ](https://github.com/henrypp/simplewall)(Due to OS I have to use it in an older version).
That firewall should have noticed me about the apps' internet requests (as it normally does) - but doesn't, not even a log entry - just blocks.

Off-topic (beware!)
@GrampaWildWilly 
Maybe you wouldn't be too "infurated" about my staying with Windows 7 (not 8). It's not because of stubborness, comfort or ignorance.
It's data safety/ privacy why _I will never use a Window beyond 7_. I did everything imaginable to keep Win7 clean (believe me). Still I'm uneasy about the situation of course.
When firefox will stop support Win7 in September.... I will change back to Linux (after some 30 years). 3 % market share and forums populated with arrogance... And all my fine developments...
Maybe you get a notion of my situation, Grampa :)
Thank you very much! :)

**Reply by GrampaWildWilly** [reply]
Why not off topic?  It's so much fun.

I hear you about going to a new OS.  I was on W7 for over 10 years but Firefox dropping support was the last straw for me.  I bit the bullet & installed W11 (an ordeal in itself, thank goodness for elevenforum) on a new computer I built.  The OS wasn't the only reason I upgraded.  My hardware was woefully antique as well.

As for Linux, I've noted that arrogance as well.  But I could see through that if the instructions were worth anything.  Every time I try to investigate the possibility of trying Linux, never mind actually making the leap to use it as my only OS, I'm hit with instructions that go something like this.  Do step 1.  Then do step 2.  Then do step 3, but before you can do that, you must have done this other thing at step 1 that we didn't tell you about then.  So I shake my head & go look up the step they forgot to tell me about earlier.  It goes once again like step 1, step 2, step 3 but oh there's something we didn't tell you about earlier.  I've never found proper instructions that go in a straight line, no switchbacks, no detours, just a simple straight line.  Another thing that has put me off is that practically the first thing I get is a scholarly dissertation on the directory structure.  No.  I don't need to know what each directory is for before you've even told me how to create a file, how to save a file, how to find a file, how to open it again, how to install software of my choosing.  You can tell me all about the purpose of each directory LATER, after you've shown me how to navigate around in the structure, and after you've told me about things that are more important.  Linux is tempting but it's too badly documented for me to dive in.

**Reply by nji9nji9** [reply]
You might prevent the diving by carefully dipping a toe in it: dual boot, and hopefully send M$ to hell some months later. (?)

**Comment by LinuxforPunks** [comment]
I have a similar problem like the OP since installing 8.1.4.0 on Win7, with it giving the error "Error number -138 occurred"

Also I tried installing the beta version.

In my case it's affecting Waterfox G6.0.8 (64-bit) and Tor Browser 13.0.9

C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe 
C:\Program Files\DownloadHelper CoApp\ffmpeg.exe 
C:\Program Files\DownloadHelper CoApp\ffprobe.exe

are whitelisted in Windows Firewall and Avast antivirus.

Like nji9nji9 I can fall back to Linux, so if it can't work please someone let me know. In the wider world, the dropping of support for Win7 has been shambolic with lots of developers not realizing they need to communicate it. e.g. Steam had a message all year saying the software would stop working on 31/12/2023 but it hasn't, and Fujitsu bricked my brand new scanner by deleting its driver from their website. Whereas with Linux if something has an open-source driver it'll still be able to be made to work years into the future - and sometimes by me.

**Reply by nji9nji9** [reply]
You should quote the complete error text to judge if it's a connection problem like with me.

I "solved" my problem by de- and re-install my (outgoing) firewall (see above). Actually it stopped noticing _any_ new internet requests, just blocked them.
It's a miracle to me what happened. Did a bit flip on the system SSD? Strange this.

**Reply by LinuxforPunks** [reply]
SideDownload error:

[tcp @ 000000000052e9c0] Connection to tcp://manifest.googlevideo.com:443 failed: Error number -138 occurred

https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1707451731/ei/81DFZfazNJeKp-oP48-ZKA/ip/217.146.112.235/id/5430992e5b38de9d/itag/270/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/sgovp/clen%3D47306329%3Bdur%3D293.080%3Bgir%3Dyes%3Bitag%3D137%3Blmt%3D1706753293909452/rqh/1/hls_chunk_host/rr1---sn-aigzrnsz.googlevideo.com/xpc/EgVo2aDSNQ%3D%3D/mh/LA/mm/31,26/mn/sn-aigzrnsz,sn-5hnekn7k/ms/au,onr/mv/m/mvi/1/pl/20/initcwndbps/1221250/vprv/1/playlist_type/DVR/dover/13/txp/5535434/mt/1707429654/fvip/5/short_key/1/keepalive/yes/fexp/24007246/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,sgovp,rqh,xpc,vprv,playlist_type/sig/AJfQdSswRQIhAJ6JVcwqnMCpXR1Xyj7z1c_yrWqRgbGnjAWXOgwaILl3AiBpGyFlA0FTPJEMSvHBwp1dZtj7qDonr8_uVf_aTzmLCg%3D%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AAO5W4owRQIgLloRJG61Wgd6QhJa-HOJQLTULuX2hgqDdLc9JmHZJLoCIQC17Bh1H3YyqEmIpfR-yCLa_miLkP_UfHVw7sKEBcgGZA%3D%3D/playlist/index.m3u8: Error number -138 occurred

**Reply by nji9nji9** [reply]
It's surely not me to have enough knowledge to help you in detail. But if the problem should be similar than mine you maybe should try by fiddling with your firewall and with Avast (being more than just antivirus but firewall too?).
The firewall I use (see above) is kind of special, in steering/ feeding the windows firefall. Maybe you should check if your firefall effects VDH, or just blocks (as with me)... But I'm only guessing... sorry...

**Reply by paulrouget** [reply]
I agree. This is a firewall issue.

**Reply by LinuxforPunks** [reply]
I've looked at this again and strangely:- adding the rules via Start>"Allow a program through Windows Firewall" didn't work, but the Control Panel>Windows Firewall>Advanced Settings>Inbound Rules (and Outbound Rules) does work. I used to use the Malwarebytes firewall and wonder if that somehow broke the former of those two interfaces.

Or another oddity with the former interface is that it registers vdhcoapp.exe as "Node.js: Server-side JavaScript" (but the Path is correct in the Details... applet)
And another is that the former interface only pulls through the title of rules created under the latter interface, not the Path - even though those are the rules that work

**Reply by nji9nji9** [reply]
There are some indices that let me guess the underlying cause is another bug in Win7, again.
But that doesn't play any role anymore.
